server/server.go

```diff
@@ -152,6 +153,17 @@ func (s *Server) RefreshMembers() {
 	s.OrgMembers = members
 }
 
+func (s *Server) withRecovery(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if x := recover(); x != nil {
+				mlog.Error("recovered from a panic", mlog.String("url", r.URL.String()), mlog.Any("error", x))
```

Comment:
Shall we also print the request body and header? It would be great to see which request caused the panic :)

Reply:
I think that would pollute the logs too much. And plus you need another NopCloser to read the body again.

Header is unlikely to contain any helpful information. The thing to check would be the body. I am guessing that the panic trace should be enough for us to figure out the bug.

Reply:
I see, let's continue with this then. 👍